Makefile

```diff
@@ -1,11 +1,31 @@
 CXX=g++
 CXXOPTIMIZE= -O2
-CXXFLAGS= -g -Wall -pthread -std=c++11 $(CXXOPTIMIZE)
-SRCFILES = main.cpp server.cpp reply.cpp config.cc config_parser.cc
+CXXFLAGS= -g -Wall -pthread -std=c++11
+SRCFILES = server.cpp reply.cpp config.cc config_parser.cc
+GTEST_DIR = googletest/googletest
+
+all: CXXFLAGS += $(CXXOPTIMIZE)
+all: SRCFILES += main.cpp
+all: webserver
+
+test: CXXFLAGS += -isystem ${GTEST_DIR}/include
+test: SRCFILES += config_parser_test.cc config_test.cc ${GTEST_DIR}/src/gtest_main.cc libgtest.a
+test: | clean webserver_test
```

Comment:
Will definitely add that but did not yet know the class names back than. Thank you!